src/codeu/chat/server/Controller.java

```diff
@@ -34,9 +43,79 @@
   private final Model model;
   private final Uuid.Generator uuidGenerator;
 
+  private final Serializer<Collection<Message>> localMessages = Serializers.collection(Message.SERIALIZER);
+  private final Serializer<Collection<ConversationHeader>> localConversationHeaders = Serializers.collection(ConversationHeader.SERIALIZER);
+  private final Serializer<Collection<User>> localUsers = Serializers.collection(User.SERIALIZER);
+
+  private final LocalFile localFile;
+
+  private final File userFile;
+  private final File conversationFile;
+  private final File messageFile;
+
+  private boolean isInitialized = false;
+
   public Controller(Uuid serverId, Model model) {
     this.model = model;
     this.uuidGenerator = new RandomUuidGenerator(serverId, System.currentTimeMillis());
+    this.localFile = new LocalFile(new File("."));
+    this.userFile = new File(localFile.getPath() + LocalFile.USER_FILE_NAME);
+    this.conversationFile = new File(localFile.getPath() + LocalFile.CONVERSATION_FILE_NAME);
+    this.messageFile = new File(localFile.getPath() + LocalFile.MESSAGE_FILE_NAME);
+  }
+  //New constructor, which can get the local file information.
+  public Controller(Uuid serverId, Model model,LocalFile localFile) {
+    this.model = model;
+    this.uuidGenerator = new RandomUuidGenerator(serverId, System.currentTimeMillis());
+    
+    this.localFile = localFile;//The path is assigned by server.
+
+    userFile = new File(localFile.getPath() + LocalFile.USER_FILE_NAME);
```

Comment:
I'd recommend moving all of this code, that reads from your save files, into the `LocalFile` class. It's usually good to group similar functionality in the same place, and I think it would be good to group all reading and writing to/from files together.